lxd/storage/backend_lxd.go

```diff
@@ -1870,7 +1870,7 @@ func (b *lxdBackend) recvVolumeFiller(conn io.ReadWriteCloser, contentType drive
 			}
 		} else {
 			// Receive block volume.
-			to, err := os.OpenFile(rootBlockPath, os.O_WRONLY|os.O_TRUNC, 0)
+			to, err := os.OpenFile(rootBlockPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0)
```

Comment:
how come this worked before without this in <= 6.1?

Also if we are creating the file now, do we need to ensure the file is created with the same permissions as a normal empty block volume file for dir pools?

Reply:
> how come this worked before without this in <= 6.1?

Because in case of `migration` we first create the volume and then receive the image into it, not using the filler as in `conversion` case.

In case of `dir` storage pool the file is created (if missing) only once the filler functions has been executed.

https://github.com/canonical/lxd/blob/2a2bead82fefcc02039a92f2a437b3db3d80670c/lxd/storage/drivers/driver_dir_volumes.go#L65-L86

File is created within `ensureVolumeBlockFile` function:
https://github.com/canonical/lxd/blob/2a2bead82fefcc02039a92f2a437b3db3d80670c/lxd/storage/drivers/utils.go#L400-L405

Reply:
> Also if we are creating the file now, do we need to ensure the file is created with the same permissions as a normal empty block volume file for dir pools?

Agree. I think 0600 should be sufficient.

Reply:
Is it the same as what is used for new volume files?

Reply:
> Because in case of migration we first create the volume and then receive the image into it, not using the filler as in conversion case.
Right makes sense thanks.